apps/files_versions/lib/Listener/FileEventsListener.php

```diff
@@ -323,20 +323,28 @@ public function pre_renameOrCopy_hook(Node $source, Node $target): void {
 			return;
 		}
 
-		// if we rename a movable mount point, then the versions don't have
-		// to be renamed
+		// if we rename a movable mount point, then the versions don't have to be renamed
 		$oldPath = $this->getPathForNode($source);
 		$newPath = $this->getPathForNode($target);
-		$absOldPath = Filesystem::normalizePath('/' . \OC_User::getUser() . '/files' . $oldPath);
+		if ($oldPath === null || $newPath === null) {
+			return;
+		}
+
+		$user = \OC_User::getUser();
+		if ($user === false) {
+			return;
+		}
+
+		$absOldPath = Filesystem::normalizePath('/' . $user . '/files' . $oldPath);
 		$manager = Filesystem::getMountManager();
 		$mount = $manager->find($absOldPath);
 		$internalPath = $mount->getInternalPath($absOldPath);
 		if ($internalPath === '' and $mount instanceof MoveableMount) {
 			return;
 		}
 
-		$view = new View(\OC_User::getUser() . '/files');
-		if ($view->file_exists($newPath)) {
+		$view = new View($user . '/files');
+		if ($view->file_exists($newPath) === true) {
```

Comment:
```suggestion
		if ($view->file_exists($newPath)) {
```
Feels dangerous to do a strong comparison here, what’s the reason?

Reply:
This was marked as resolved without an answer

Reply:
I was sure to reverted this... Anyways will do so.

But it should be save what other truethy value could be returned? At least non documented :sweat_smile: